datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/concurrency/GraphQLConcurrencyUtils.java

```diff
@@ -0,0 +1,28 @@
+package com.linkedin.datahub.graphql.concurrency;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Supplier;
+
+public class GraphQLConcurrencyUtils {
+  private GraphQLConcurrencyUtils() {}
+
+  private static ExecutorService graphQLExecutorService = null;
+
+  public static ExecutorService getExecutorService() {
+    return GraphQLConcurrencyUtils.graphQLExecutorService;
+  }
+
+  public static void setExecutorService(ExecutorService executorService) {
+    GraphQLConcurrencyUtils.graphQLExecutorService = executorService;
+  }
+
+  public static <T> CompletableFuture<T> supplyAsync(Supplier<T> supplier) {
```

Comment:
Can we require a name to be passed here so we can emit metrics about the source of the tasks? Maybe the name of the calling class?

Reply:
Do you have specific metrics you'd like me to go ahead and add? I'd rather not add in arguments for future use. So we're proposing a:

* Counter metric based on class name for number of executed supplyAsync calls, something like <ClassName>_graphQL_async_execution[_count] ?

Any others?

Reply:
Proposal for now is a simple counter of the "source" of async tasks. I'm thinking `supplyAsync` should take `String simpleClassName` and `String taskName` (both of which can be null) and we increment a counter.

```
MetricUtils.counter(MetricRegistry.name(this.getClass(),  "supplyAsync", simpleClassName, taskName)).inc();
```

Which for a `supplyAsync` call from `EntityBrowsePathsResolver` would produce the metric:

`metrics_com_..._GraphQLConcurrencyUtils_supplyAsync_EntityBrowsePathsResolver_taskName`

Note that a requirement here is to have a common prefix so we can, for example, look up all metrics with the `*.supplyAsync.*` pattern to see the counts per task or sum them all up to get total count. Not sure if having `GraphQLConcurrencyUtils` in there makes sense, but we just need a common prefix.

Reply:
Also if `taskName` is omitted, the metric would look like:

```
metrics_com_..._GraphQLConcurrencyUtils_supplyAsync_EntityBrowsePathsResolver
```

which is fine. It exists for cases where there might be multiple different `supplyAsync` from the same class.